plugins/core/gs-appstream.c

```diff
@@ -765,6 +765,10 @@ gs_appstream_refine_app (GsPlugin *plugin,
 	    as_app_get_language (item, tmp) > 50)
 		gs_app_add_kudo (app, GS_APP_KUDO_MY_LANGUAGE);
 
+	/* Add popular tag to com.endlessm. tags See discussion on T23152 */
+	if (g_str_has_prefix (gs_app_get_id (app), "com.endlessm."))
+		as_app_add_kudo (item, "GnomeSoftware::popular");
```

Comment:
Actually, I think this would be best if it was done in `gs_plugin_refine()` rather than `gs_plugin_add_popular()`, but either way I think you should just jump straight to `GS_APP_KUDO_FEATURED_RECOMMENDED` instead of trying to get it managed through appstream.

Reply:
Yeah, the add_popular is the wrong place for this if we want to do it in the eos plugin, because it's only passed the other apps in the list if they are already popular. If every plugin's refine is called on every app, that works.